src/app/common/validation/forms/fee-validators.ts

```diff
@@ -0,0 +1,44 @@
+import BigNumber from 'bignumber.js';
+import { RequiredNumberSchema } from 'yup/lib/number';
+import { AnyObject } from 'yup/lib/object';
+
+import { BTC_DECIMALS, STX_DECIMALS } from '@shared/constants';
+import { isNumber } from '@shared/utils';
+
+import { formatInsufficientBalanceError, formatPrecisionError } from '@app/common/error-formatters';
+import {
+  btcAmountValidator,
+  stxAmountValidator,
+} from '@app/common/validation/forms/currency-validators';
+
+import { btcToSat, stxToMicroStx } from '../../money/unit-conversion';
+
+function feeValidatorFactory(
+  symbol: string,
+  precision: number,
+  unitConverter: (unit: string | number | BigNumber) => BigNumber,
+  validator: (errorMsg: string) => RequiredNumberSchema<number | undefined, AnyObject>,
+  availableBalance?: BigNumber
+) {
```

Comment:
Nice way to reuse validation logic. 

Were availableBalance passed in as `Money`, we'd not need both `symbol` and `availableBalance`. API wise, with this many arguments, and order being unimportant, I'd recommend it being an object rather than sequence of params.

Reply:
Good point here, thx for pointing out. Will fix.